worklist/views.py

```diff
@@ -189,23 +189,27 @@ def search_task(request, username, worklist_created_by, worklist_name):
 
 
 # View to update the table displaying all tasks of a worklist
-def update_task_table(request, worklist_created_by, worklist_name):
+@with_logged_in_user
+def update_task_table(request, username, worklist_created_by, worklist_name):
     search_term = request.GET.get('search_term', '')
     results = search_task_helper(search_term, worklist_name, worklist_created_by)
 
-    return render(request, 'show-task-table.html', {'tasks': results['tasks']})
+    return render(request, 'show-task-table.html', {'tasks': results['tasks'],
+                                                    'logged_in_user': username})
 
 
 # Update task information when user edits it
 @login_required
-def update_task_info(request, worklist_created_by, worklist_name):
+def update_task_info(request):
     username = request.user.username
+    worklist_created_by = request.POST.get('worklist_created_by', '')
+    worklist_name = request.POST.get('worklist_name', '')
     article_name = request.POST.get('article_name', '')
     status = request.POST.get('status', '')
     progress = request.POST.get('progress', '')
     claimed_by = ''
 
-    if status == 'claimed':
+    if status == 'Claimed':
```

Comment:
Consider turning the status and progress strings into constants. That avoids typos and makes it easier to change the actual strings later if we have to.

Reply:
Done.